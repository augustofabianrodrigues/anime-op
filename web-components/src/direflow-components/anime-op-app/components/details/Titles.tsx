import React, { FC } from 'react';
import LanguageDictionaryList from '../shared/LanguageDictionaryList';
import RegularSection from './RegularSection';

interface TitlesProps {
  titles: {
    [lang: string]: string;
  };
}

const Titles: FC<TitlesProps> = ({ titles }) => {
  return (
    <RegularSection title="Titles" className="titles">
      <LanguageDictionaryList values={titles} />
    </RegularSection>
  );
};

export default Titles;
