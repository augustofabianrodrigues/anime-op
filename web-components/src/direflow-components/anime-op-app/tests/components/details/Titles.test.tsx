import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Titles from '../../../components/details/Titles';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Titles
      titles={{
        en: 'Cowboy Bebop',
        en_jp: 'Cowboy Bebop',
        ja_jp: 'カウボーイビバップ',
      }}
    />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('works without titles', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Titles titles={{}} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('displays titles that are passed', () => {
  const titles: { [lang: string]: string } = {
    en: 'Cowboy Bebop',
    en_jp: 'Cowboy Bebop',
    ja_jp: 'カウボーイビバップ',
  };

  const wrapper = shallow(<Titles titles={titles} />);

  for (const lang in titles) {
    const node = wrapper.findWhere((el) => el.key() === lang);
    expect(node).toExist();

    expect(node).toIncludeText(lang);
    expect(node).toIncludeText(titles[lang]);
  }
});

it('snapshot has only a "[en] Cowboy Bebop" item', () => {
  const renderTree = renderer
    .create(<Titles titles={{ en: 'Cowboy Bebop' }} />)
    .toJSON();
  expect(renderTree).toMatchSnapshot();
});
