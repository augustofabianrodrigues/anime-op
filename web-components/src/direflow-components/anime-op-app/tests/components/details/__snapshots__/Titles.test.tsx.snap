// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot has only a "[en] Cowboy Bebop" item 1`] = `
<section
  className="synopsis regular-section titles"
>
  <h3
    className="section-title"
  >
    Titles
  </h3>
  <ul
    className="titles-list"
  >
    <li>
      <span
        className="lang"
      >
        en
      </span>
      Cowboy Bebop
    </li>
  </ul>
</section>
`;
